frontend/src/Components/ProviderHomePage/ProviderHomePage.jsx

```diff
@@ -101,6 +147,39 @@ const ProviderHomePage = () => {
                 <p>{profile.bio}</p>
                 <button onClick={handleProfileUpdate}>Edit Bio</button>
             </section>
+            <section>
+                <h2>Your Schedules</h2>
+                {Object.keys(schedules).length > 0 ? (
+                    <ul>
+                        {Object.keys(schedules).map(date => (
+                            <li key={date}>
+                                <strong>{date}:</strong>
+                                <ul>
+                                    {schedules[date].slots?.map((slot, index) => (
+                                        <li key={index}>
+                                            {new Date(slot).toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' })}
+                                        </li>
+                                    ))}
+                                </ul>
+                            </li>
+                        ))}
+                    </ul>
+                    ) : (
+                    <p>You have no schedules set up.</p>
+                )}
+                <button onClick={() => setShowModal(true)}>Set Up Schedule</button>
+            </section>
+            {showModal=== true && <ScheduleModal
```

Comment:
Do we need the `showModal=== true` here is we have it inside the `ScheduleModal` component? Would be good to check if it works with just one of them.

Reply:
I will update this as well